server/relay_entry.go

```diff
@@ -41,8 +42,19 @@ func NewRelayEntry(relayURL string) (entry RelayEntry, err error) {
 		return entry, ErrMissingRelayPubkey
 	}
 
+	// Convert the username string to a public key.
 	err = entry.PublicKey.UnmarshalText([]byte(entry.URL.User.Username()))
-	return entry, err
+	if err != nil {
+		return entry, err
+	}
+
+	// Check if the public key is the point-at-infinity.
+	pointAtInfinity := [48]byte{}
```

Comment:
worth having this as a constant to avoid allocating each call?

Reply:
Sure, I'm good with that. Fixed.